Zend/zend_closures.c

```diff
@@ -381,7 +381,45 @@ static ZEND_COLD zend_function *zend_closure_get_constructor(zend_object *object
 static int zend_closure_compare(zval *o1, zval *o2) /* {{{ */
 {
 	ZEND_COMPARE_OBJECTS_FALLBACK(o1, o2);
-	return Z_OBJ_P(o1) != Z_OBJ_P(o2);
+
+	zend_closure *lhs = (zend_closure*) Z_OBJ_P(o1);
+	zend_closure *rhs = (zend_closure*) Z_OBJ_P(o2);
+
+	if (lhs == rhs) {
+	    return SUCCESS;
+	}
```

Comment:
You can drop this check. It was unnecessary in the original code as well (the case where o1 and o2 are the exact same object is handled at a higher level).

Reply:
Fixed in 26b3ee27611dff0062831dc1e77315bcc9d3acea